### ISSUE TITLE
`az login` fails with "ssl.SSLEOFError: EOF occurred in violation of protocol"

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#19456]
## Symptom

Customer reported issue:

`az login` fails while running Azure CLI 2.25.0. I am using CLI behind proxy and followed the guidance from https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively?view=azure-cli-latest#work-behind-a-proxy.

I have added the root CA certificate to `cacert.pem` file and also set up environment variables. The setup is working as expected in cli module 2.24.0

```
$ az login --debug

Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/connectionpool.py", line 696, in urlopen
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/connectionpool.py", line 964, in _prepare_proxy
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/connection.py", line 359, in connect
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/connection.py", line 500, in _connect_tls_proxy
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
  File "ssl.py", line 500, in wrap_socket
  File "ssl.py", line 1040, in _create
  File "ssl.py", line 1309, in do_handshake
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1125)
...
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/profile/custom.py", line 180, in login
knack.util.CLIError: Certificate verification failed. This typically happens when using Azure CLI behind a proxy that intercepts traffic with a self-signed certificate. Please add this certificate to the trusted CA bundle. More info: https://docs.microsoft.com/cli/azure/use-cli-effectively#work-behind-a-proxy. Error detail: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /common/oauth2/token (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1125)')))
```

IcM: 252915638

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
For error:

> Max retries exceeded with url: /common/oauth2/token (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1125)')))

We bumped `urllib3` from 1.25.9 to 1.26.4 in Azure CLI 2.25.0 (https://github.com/Azure/azure-cli/pull/18213). Per the change log of `urllib3`:

https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#1260-2020-11-10

> 1.26.0 (2020-11-10)
> Added support for HTTPS proxies contacting HTTPS servers

Before this change, `HTTPS_PROXY=https://xxx:xxx` will be treated by `urllib3` as `http://` even if the URL specifies `https://`. Since 1.26.0, `https://` is honored by `urllib3` (https://github.com/urllib3/urllib3/issues/1850).

Therefore, `HTTPS_PROXY` environment variable should be set according to the real proxy configuration. If it is an HTTP proxy, `HTTPS_PROXY` should point to `http://`, instead of `https://`.

References:

- https://github.com/pypa/pip/issues/9216#issuecomment-741836058
- https://github.com/psf/requests/issues/5740

**Comment by dan-dimitrov** [comment]
We have just ran into the same error message during "az login". We used "az login --debug" and realised that despite the same ssl error message, the device was unable to hit management.azure.com:443 due to our firewall rules. We wasted quite some time verifying certificate chains to make sure that our Azure Firewall without TLS inspection was not changing the certificate chain, as the error message set us astray.
We went in and made sure that all Azure CLI endpoints were enabled as per the docs (https://learn.microsoft.com/en-us/cli/azure/azure-cli-endpoints?tabs=azure-cloud), and we were able to authenticate successfully.
